From 0168ff83feb2a635ed60f948e662df662af7454d Mon Sep 17 00:00:00 2001
From: Karthik Ramanan <a0393906@ti.com>
Date: Mon, 2 Dec 2013 16:06:17 +0530
Subject: [PATCH] weston: handle touch when keyboard not connected

If the keyboard is not connected, the seat for the keyboard
will not be initialized. When the touch event occurs, the
touch release event notifies the keyboard handler.
The handler tries to handle access the keyboard related
information. The patch fixes the issue by checking for the
keyboard registration prior to the access.

Note that to verify this issue, we have to reboot the board
without the keyboard or mouse connected and then invoke
the touch screen.

Signed-off-by: Karthik Ramanan <a0393906@ti.com>
---
 src/compositor.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/compositor.c b/src/compositor.c
index ced3883..e226734 100644
--- a/src/compositor.c
+++ b/src/compositor.c
@@ -1730,6 +1730,10 @@ notify_key(struct weston_seat *seat, uint32_t time, uint32_t key,
 	   enum weston_key_state_update update_state)
 {
 	struct weston_compositor *compositor = seat->compositor;
+
+	if(!(seat->has_keyboard))
+             return;
+
 	struct wl_keyboard *keyboard = seat->seat.keyboard;
 	struct weston_surface *focus =
 		(struct weston_surface *) keyboard->focus;
-- 
1.7.0.4

